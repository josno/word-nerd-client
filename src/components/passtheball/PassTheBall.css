img {
	height: 20vh;
}

.game-play-container {
	height: 400px;
}
