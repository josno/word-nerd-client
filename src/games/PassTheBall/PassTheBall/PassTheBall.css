.ball-gif {
	width: 100%;
	width: 150px;
	height: auto;
}

.ball-image-container {
	background-color: #18c0f0;
}

@media (min-width: 900px) {
	.ball-gif {
		width: 400px;
	}
}
