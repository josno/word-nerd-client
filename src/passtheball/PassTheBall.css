.pass-the-ball {
	margin-top: 20vh;
}

img {
	height: 20vh;
}
