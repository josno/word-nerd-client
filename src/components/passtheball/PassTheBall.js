import React, { Component } from 'react';
import Ball from './ball.png';
import './PassTheBall.css';
import Sound from 'react-sound';
import soundfile from './pass-song.mp3';
import PassTheBallMessage from '../PassTheBallMessage/PassTheBallMessage';

class PassTheBall extends Component {
	constructor(props) {
		super(props);
		this.state = {
			error: null
		};
	}

	render() {
		return (
			<div className="game-play-container">
				<PassTheBallMessage />
				<Sound
					url={soundfile}
					playStatus={Sound.status.PLAYING}
					onFinishedPlaying={this.handleSongFinishedPlaying}
				/>
				<img alt="ball-placeholder" src={Ball} />
			</div>
		);
	}
}

export default PassTheBall;
