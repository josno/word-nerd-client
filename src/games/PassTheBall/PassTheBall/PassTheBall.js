import React, { Component } from "react";
import Ball from "./giphy.gif";
import "./PassTheBall.css";
import Sound from "react-sound";
import soundfile from "./pass-song.mp3";
import HeaderMessage from "../../../components/HeaderMessage/HeaderMessage";

class PassTheBall extends Component {
	render() {
		return (
			<section className='game-play-container'>
				<HeaderMessage oldMessage='Start!' newMessage='Pass The Ball' />
				<Sound
					url={soundfile}
					playStatus={Sound.status.PLAYING}
					onFinishedPlaying={this.handleSongFinishedPlaying}
				/>
				<div className='ball-image-container'>
					<img className='ball-gif' alt='ball-placeholder' src={Ball} />
				</div>
			</section>
		);
	}
}

export default PassTheBall;
