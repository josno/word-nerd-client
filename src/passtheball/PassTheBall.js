import React from 'react';
import Ball from './ball.png';
import './PassTheBall.css';

export default function PassTheBall() {
	return (
		<div className="pass-the-ball">
			<nav role="navigation">nav</nav>
			<h1 class="pass-the-ball">Pass The Ball!</h1>
			<p>
				<img alt="ball-placeholder" src={Ball} />
			</p>
			<p>
				<h2>(Timer goes here)</h2>
			</p>
			<div class="game-controls">
				<a href="eight-end-game.html">
					<button class="game-end">End Game</button>
				</a>
				<a href="six-guess-the-word.html">
					<button class="next-button">Next</button>
				</a>
			</div>
		</div>
	);
}
