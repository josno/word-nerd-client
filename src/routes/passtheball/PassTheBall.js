import React from 'react';
import Ball from './ball.png';
import { Link } from 'react-router-dom';
import './PassTheBall.css';

export default function PassTheBall() {
	return (
		<div className="game-play-container">
			<nav role="navigation">nav</nav>
			<h1 className="pass-the-ball">Pass The Ball!</h1>

			<img alt="ball-placeholder" src={Ball} />

			<h2>(Timer goes here)</h2>

			{/* need to be component */}
			<div className="game-controls">
				<Link to="/end-game-page">
					<button className="game-end">End Game</button>
				</Link>
				<Link to="/guess-the-word">
					<button className="next-button">Next</button>
				</Link>
			</div>
		</div>
	);
}
