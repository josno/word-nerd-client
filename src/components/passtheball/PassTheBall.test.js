import React from './node_modules/react';
// import ReactDOM from 'react-dom';
import { BrowserRouter } from './node_modules/react-router-dom';
import renderer from './node_modules/react-test-renderer';
import PassTheBall from './PassTheBall';

describe('<PassTheBall />', () => {
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<PassTheBall />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
