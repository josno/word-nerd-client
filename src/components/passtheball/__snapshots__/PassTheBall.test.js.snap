// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PassTheBall /> renders as expected 1`] = `
<div
  className="game-play-container"
>
  <div
    className="pass-the-ball"
  >
    <h1>
      Start
    </h1>
  </div>
  <img
    alt="ball-placeholder"
    src="ball.png"
  />
</div>
`;
